x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/parser/promql/PromqlLogicalPlanBuilder.java

```diff
@@ -432,12 +432,7 @@ public LogicalPlan visitFunction(PromqlBaseParser.FunctionContext ctx) {
         Source source = source(ctx);
         String name = ctx.IDENTIFIER().getText().toLowerCase(Locale.ROOT);
 
-        Boolean exists = PromqlFunctionRegistry.INSTANCE.functionExists(name);
-        if (Boolean.TRUE.equals(exists) == false) {
-            String message = exists == null ? "Function [{}] not implemented yet" : "Unknown function [{}]";
-            throw new ParsingException(source, message, name);
-        }
-
+        PromqlFunctionRegistry.INSTANCE.checkFunction(source, name);
         var metadata = PromqlFunctionRegistry.INSTANCE.functionMetadata(name);
```

Comment:
Now that I see this, can we simplify to throw the exception in `functionMetadata` if it gets a function name that doesn't exist?